### PR TITLE
DP-22929 Accessibility fixes for pages within binders

### PR DESCRIPTION
**Pull request by ygannett**
**Description:**

**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-22929

**To Test:**
- [ ] Build the site with Mayflower, https://github.com/massgov/mayflower/pull/1433/files.
- [ ] Go to a binder page. ex) http://mass.local/rules-of-civil-procedure/civil-procedure-rule-27-depositions-before-action-or-pending-appeal
- [ ] Tab to navigate the table of contents overlay open button.
- [ ] Hit ENTER to open the overlay.
- [ ] Tab through focusable items to the last one, then hit TAB again.  Find the close (x) button gets focus. Focus doesn't move onto the main page elements. As long as the overlay is open, focus loops in the overlay.
- [ ] While the overlay is open, hit ESC key.  It doesn't matter where focus is in the overlay.  Find the ESC key triggers to close the overlay and focus is on the overlay open button.
- [ ] Tab through the page flipper component. Find there are only 3 items, this is part of, previous and next page links get focus.  The arrow buttons, which are linked to the previous and next pages as duplicates, don't get focus. Mouse users can still click the button links to go to their linked pages.
- [ ] With inspect, check each page link's `aria-label`.  Find value contains contextual info:
      - **This page is part of** PAGE TITLE.
      - **Previous page,** PAGE TITLE,
      - **Next page,** PAGE TITLE
- [ ] In screen width 620px and smaller, check the markup of the arrow buttons in inspect.  It has `tabindex="0"` and no `aria-hidden="true"`. With keyboard and screen readers, the button gets focus. The text label beside the buttons shown in screen size 621px and greater are now hidden with css, `display: none;`.

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: docroot/themes/custom/mass_theme/templates/field/extra-node-pager.html.twig [hunk 1]
```diff
@@ -6,44 +6,51 @@
...
 <div class="referencing-binders" data-id="{{ parent.entity.nid.value }}">
 {% if next %}
+  {% set ariaLabel = "Next page, " ~ next['title'] %}
   {% set next_button = {
     "arrowButton": {
       "href": next['url'],
       "info": "",
-      "direction": "right"
+      "direction": "right",
+      "ariaHide": TRUE
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/templates/field/extra-node-pager.html.twig]:
```suggestion
      "ariaHidden": TRUE
```

File: docroot/themes/custom/mass_theme/templates/field/extra-node-pager.html.twig [hunk 2]
```diff
@@ -6,44 +6,51 @@
...
 {% if previous %}
+  {% set ariaLabel = "Previous page, " ~ next['title'] %}
   {% set prev_button = {
     "arrowButton": {
       "href": previous['url'],
-      "info": "",
-      "direction": "left"
+      "info": ariaLabel,
+      "direction": "left",
+      "ariaHidden": TRUE
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/templates/field/extra-node-pager.html.twig]:
```suggestion
      "ariaHidden": true
```

File: docroot/themes/custom/mass_theme/templates/field/extra-node-pager.html.twig [hunk 3]
```diff
@@ -6,44 +6,51 @@
...
 {% if next %}
+  {% set ariaLabel = "Next page, " ~ next['title'] %}
   {% set next_button = {
     "arrowButton": {
       "href": next['url'],
-      "info": "",
-      "direction": "right"
+      "info": ariaLabel,
+      "direction": "right",
+      "ariaHidden": TRUE
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/templates/field/extra-node-pager.html.twig]:
```suggestion
      "ariaHidden": true
```

File: docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  var tocFocusableElements = $('.js-inline-overlay').find(':focusable');
+  var firstFocusableElement = tocFocusableElements[0];
+  var lastFocusableElement = tocFocusableElements[tocFocusableElements.length - 1];
+  var tocButtons = $('.ma__toc--hierarchy__container .js-accordion-link');
+  var tocLastButton = tocButtons[tocButtons.length - 1];
+
+  $(tocLastButton).keydown(function (e) {
+    var key = (e.keyCode ? e.keyCode : e.witch);
+    var lastContainer = $(tocLastButton).closest('.ma__toc--hierarchy__accordion');
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js]:
```suggestion
    var lastContainer = $(tocLastButton).closest('.ma__toc--hierarchy__accordion');
```
can you move this out of the keydown function

File: docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  'use strict';
+
+  var tocFocusableElements = $('.js-inline-overlay').find(':focusable');
+  var firstFocusableElement = tocFocusableElements[0];
+  var lastFocusableElement = tocFocusableElements[tocFocusableElements.length - 1];
+  var tocButtons = $('.ma__toc--hierarchy__container .js-accordion-link');
+  var tocLastButton = tocButtons[tocButtons.length - 1];
+
+  $(tocLastButton).keydown(function (e) {
+    var key = (e.keyCode ? e.keyCode : e.witch);
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js]:
```suggestion
    var key = e.keyCode || e.which;
```

File: docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    if (!lastContainer.hasClass('is-open')) {
+      // When tab key is hit on the last ficusable element,
+      if (key === '9') {
+        // Set focus on the first focusable element in the overlay.
+        firstFocusableElement.focus();
+      }
+    }
+
+    // When the sub menu is open.
+    if (lastContainer.hasClass('is-open')) {
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js]: I think you could pull this block out of the parent function `$(tocLastButton).keydown(function (e) {...}`

File: docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      $(lastFocusableElement).keydown(function (e) {
+        var key = (e.keyCode ? e.keyCode : e.witch);
```
Comment (clairesunstudio) [inline docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js]:
```suggestion
        var key = e.keyCode || e.which;
```

Comment (clairesunstudio) [reply docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js]: oh do you mean `event.which`?

File: docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js (hunk 4 above)
Comment (ygannett) [reply docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js]: The change is made as suggested.

File: docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js (hunk 6 above)
Comment (ygannett) [reply docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js]: The block is moved out of the function.  With the change, it still functions as intended.

File: docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js (hunk 7 above)
Comment (ygannett) [reply docroot/themes/custom/mass_theme/overrides/js/inline-overlay.js]: Yes, it's `event.which`, but it's defined as `e` in the event listener at line 24.

**Review by clairesunstudio — APPROVED** [review]

**Comment by clairesunstudio** [comment]
LGTM! Since the release tag of 0.29.0 has been cut, I will merge this into develop. Thanks @ygannett !

**Pull request merged**